### ISSUE TITLE
Make Boa errors nice

### ISSUE THREAD

**Issue by lastmjs** [demergent-labs/azle#354]
Any errors coming back from Boa should be short human-readable strings.

Here's code that should work to make the errors received from the boa context nicer:

```typescript
function getBody(implItemMethod: ImplItemMethod, inputs: any[]): Rust {
    const returnValueHandler: Rust = generateReturnValueHandler(implItemMethod);

    const functionName = implItemMethod.ident;

    return `
        fn dummy() {
            unsafe {
                let mut boa_context = BOA_CONTEXT_OPTION.as_mut().unwrap();

                let exports_js_value = boa_context.eval("exports").unwrap();
                let exports_js_object = exports_js_value.as_object().unwrap();

                let ${functionName}_js_value = exports_js_object.get("${functionName}", &mut boa_context).unwrap();
                let ${functionName}_js_object = ${functionName}_js_value.as_object().unwrap();
            
                let _azle_return_value_result = ${functionName}_js_object.call(
                    &boa_engine::JsValue::Null,
                    &[
                        ${inputs.map((input) => {
                            return `${input.typed.pat.ident.ident}.azle_into_js_value(&mut boa_context)`;
                        }).join(',')}
                    ],
                    &mut boa_context
                );

                match _azle_return_value_result {
                    Ok(_azle_return_value) => {
                        ${returnValueHandler}
                    },
                    Err(err_js_value) => {
                        let err_js_object = err_js_value.as_object().unwrap();
                        let name_js_value = err_js_object.get("name", boa_context).unwrap();
                        let message_js_value = err_js_object.get("message", boa_context).unwrap();

                        ic_cdk::api::trap(&format!(
                            "{name}: {message}",
                            name = name_js_value.as_string().unwrap().to_string(),
                            message = message_js_value.as_string().unwrap().to_string()
                        ))
                    }
                }
            }
        }
    `;
}
```

**Comment by lastmjs** [comment]
We'll probably write a Trait on the Boa Result Err type that will do this functionality for us.

**Comment by dansteren** [comment]
Just remember that when "catching" errors from JS, it's possible that the JS would have thrown a string instead of an object. We are making the assumption that it's an object, so we should make sure to handle both, rather than just doing `err_js_value.as_object().unwrap();`

**Comment by bdemann** [comment]
resolved with #717
